shared/reducers/chat.js

```diff
@@ -19,7 +19,7 @@ function _filterMessages (seenMessages: Set<any>, messages: List<ServerMessage>
   const updatedMessages = messages.map(m => messagesToUpdate.has(m.key) ? messagesToUpdate.get(m.key) : m)
   // We have to check for m.messageID being falsey and set.has(undefined) is true!. We shouldn't ever have a zero messageID
   const nextMessages = filteredPrepend.concat(updatedMessages, filteredAppend).filter(m => !m.messageID || !deletedIDs.has(m.messageID))
-  const nextSeenMessages = Set(nextMessages.map(m => m.key))
+  const nextSeenMessages = Set(nextMessages.map(m => m.key)).union(nextMessages.map(m => m.messageID).filter(x => x))
```

Comment:
lets also explicitly add outbox id in case we change out key works. i'm worried that code changes and then we don't add this to the filter and we still get dupes. Also you should be able to do filter(Boolean) instead of the x=> x